src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,617 @@
+  (i : ℕ) : measurable_set $ measure_of_nonpos_seq s f i :=
+begin
+  by_cases ¬ 0 ≤ (s ∘ f) i,
+  { simp_rw [measure_of_nonpos_seq, if_pos h],
+    exact hf i },
+  { simp_rw [measure_of_nonpos_seq, if_neg h],
+    exact measurable_set.empty }
+end
+
+lemma measure_of_nonpos_seq_of_disjoint (hf : pairwise (disjoint on f)) :
+  pairwise $ disjoint on (s.measure_of_nonpos_seq f) :=
+begin
+  rintro i j hij x ⟨hx₁, hx₂⟩,
+  by_cases hi : ¬ 0 ≤ (s ∘ f) i,
+  { by_cases hj : ¬ 0 ≤ (s ∘ f) j,
+    { simp_rw [measure_of_nonpos_seq, if_pos hi] at hx₁,
+      simp_rw [measure_of_nonpos_seq, if_pos hj] at hx₂,
+      exact hf i j hij ⟨hx₁, hx₂⟩ },
+    { simp_rw [measure_of_nonpos_seq, if_neg hj] at hx₂,
+      exact false.elim hx₂ } },
+  { simp_rw [measure_of_nonpos_seq, if_neg hi] at hx₁,
+    exact false.elim hx₁ }
+end
+
+lemma measure_of_seq_eq (i : ℕ) :
+  ∥s (f i)∥ = s (s.measure_of_nonneg_seq f i) - s (s.measure_of_nonpos_seq f i) :=
+begin
+  rw [measure_of_nonneg_seq, measure_of_nonpos_seq],
+  by_cases 0 ≤ (s ∘ f) i,
+  { have : ¬¬0 ≤ (s ∘ f) i := not_not.2 h,
+    simp_rw [real.norm_of_nonneg h, if_pos h, if_neg this,
+             s.measure_of_empty, _root_.sub_zero] },
+  { simp_rw [real.norm_of_neg (lt_of_not_ge h), if_pos h, if_neg h,
+             s.measure_of_empty, _root_.zero_sub] }
+end
+
+/-- Given a signed measure `s`, `s ∘ f` is summable for all sequence
+`f` of pairwise disjoint measurable sets. -/
+theorem summable_measure_of
+  (hf₁ : ∀ i, measurable_set (f i)) (hf₂ : pairwise (disjoint on f)) :
+  summable (s ∘ f) :=
+begin
+  apply summable_of_summable_norm,
+  simp_rw measure_of_seq_eq,
+  apply summable.sub,
+  { apply summable_measure_of_nonneg,
+    { exact measure_of_nonneg_seq_of_measurable_set hf₁ },
+    { exact measure_of_nonneg_seq_of_disjoint hf₂ },
+    { exact measure_of_nonneg_seq_nonneg } },
+  { apply summable_measure_of_nonpos,
+    { exact measure_of_nonpos_seq_of_measurable_set hf₁ },
+    { exact measure_of_nonpos_seq_of_disjoint hf₂ },
+    { exact measure_of_nonpos_seq_nonpos } },
+end
+
+/-- Obtain a `signed_measure` by constructing a `signed_measure'`. -/
+def to_signed_measure (s : signed_measure' α) : signed_measure α :=
+{ measure_of := s,
+  empty := s.empty,
+  not_measurable := s.not_measurable,
+  m_Union := λ _ hf₁ hf₂,
+    (summable.has_sum_iff (signed_measure'.summable_measure_of hf₁ hf₂)).2
+      (s.measure_of_disjoint_Union hf₁ hf₂).symm }
+
+/-- The naturally induced `signed_measure'` from a `signed_measure`. -/
+def of_signed_measure (s : signed_measure α) : signed_measure' α :=
+{ measure_of := s,
+  empty := s.empty,
+  not_measurable := s.not_measurable,
+  m_Union := λ _ hf₁ hf₂, s.measure_of_disjoint_Union hf₁ hf₂ }
+
+end signed_measure'
+
+namespace vector_measure
+
+/-- A finite measure coerced into a real function is a signed measure. -/
+def of_measure (μ : measure α) [hμ : finite_measure μ] : signed_measure α :=
```

Comment:
It is probably better to define this in the `measure` namespace, and call it `to_signed_measure`, to be able to use the dot notation like `μ.to_signed_measure`.

Reply:
With the disclaimer that I know almost no measure theory, Is the following statement true?
```suggestion
def of_measure (μ : measure α) [hμ : finite_measure μ] : vector_measure α ℝ≥0∞ :=
```

Reply:
I believe so since `ℝ≥0∞` is a `add_comm_monoid`

Reply:
We can do something like this: 
```lean
def of_measure' (μ : measure α) : vector_measure α ℝ≥0∞ := 
{ measure_of := λ i : set α, if measurable_set i then μ i else 0,
  empty := by simp [μ.empty],
  not_measurable := λ _ hi, if_neg hi,
  m_Union := λ _ hf₁ hf₂, 
  begin
    rw summable.has_sum_iff ennreal.summable,
    { rw [if_pos (measurable_set.Union hf₁), measure_theory.measure_Union hf₂ hf₁],
      exact tsum_congr (λ n, if_pos (hf₁ n)) },
  end }
```
Is this worth having?

Reply:
I think so, if only to make it clear that all measures are vector measures

Reply:
It should be called `measure.to_vector_measure` probably.

Reply:
Does a similar construction work for outer_measure?

Reply:
I don't think so since outer measures are not sigma additive